android/src/main/kotlin/im/nfc/flutter_nfc_kit/FlutterNfcKitPlugin.kt

```diff
@@ -113,14 +124,14 @@ class FlutterNfcKitPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
                 val timeout = call.argument<Int>("timeout")!!
                 // technology and option bits are set in Dart code
                 val technologies = call.argument<Int>("technologies")!!
-                thread {
+                nfcHandler.post {
```

Comment:
It would be better to check the return value of `nfcHandler.post` on each call and report error back to Flutter when getting `false`, otherwise Flutter code might get stuck on `await`.

My suggestion is defining something like
```kotlin
fun runOnNfcThread(fn: () -> Unit) {
    if (!nfcHandler.post(fn)) { result.error("500", "Failed to post job to NFC Handler thread.", null); }
}
```
in `handleMethodCall` and replace all occurrences of `nfcHandler.post`.